src/main/java/eu/europeana/downloads/ListRecordsQuery.java

```diff
@@ -90,57 +78,48 @@ private void initThreadPool() {
                 .newFixedThreadPool(threads);
     }
 
-
     @Override
     public String getVerbName() {
-        return useGetRecord ? "ListIdentifiers" : "ListRecords";
+        return "ListRecords";
     }
 
     @Override
     public void execute(OAIPMHServiceClient oaipmhServer) throws OaiPmhException {
         if (sets.isEmpty()) {
-            execute(oaipmhServer, null);
+            executeMultithreadListRecords(oaipmhServer);
         } else {
             for (String setIdentifier : sets) {
-                execute(oaipmhServer, setIdentifier);
+                executeListRecords(oaipmhServer, setIdentifier);
```

Comment:
Same as for ListIdentifiers. Would be nice if we can support this also when specifying multiple sets